backend/src/Notifo.Domain/Channels/MobilePush/UserNotificationExtensions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// ==========================================================================
+//  Notifo.io
+// ==========================================================================
+//  Copyright (c) Sebastian Stehle
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+using FirebaseAdmin.Messaging;
+using Notifo.Domain.UserNotifications;
+
+namespace Notifo.Domain.Channels.MobilePush
+{
+    public static class UserNotificationExtensions
+    {
+        public static Message ToFirebaseMessage(this UserNotification notification, string token)
+        {
+            var message = new Message
+            {
+                Token = token
+            };
+
+            var commonData = new Dictionary<string, string>();
+            commonData.AddIfPresent(() => notification.TrackingUrl);
```

Comment:
It is elegant but very slow. Please use an approach without expressions. You can just use `nameof(...)`